### ISSUE TITLE
The `never` option of `declaration-block-trailing-semicolon` takes away the semicolon Less needs to read a value its expression reader refuses

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#688]
### What minimal example or steps are needed to reproduce the bug?

Less reads an ordinary declaration's value two ways. `anonymousValue()` is tried first and matches `` /^([^.#@$+/'"*`(;{}-]*);/ `` — a run holding none of those characters **followed by a semicolon** — and the value is kept whole, unread. Only where that match fails does the expression reader `value()` run, and where it too fails the declaration is refused. So the semicolon is what saves every value the expression reader cannot read, and `never` of `declaration-block-trailing-semicolon` takes it away:

```less
a {
	color: pink !IMPORTANT;
}
```

comes back from `--fix` with the semicolon gone, and `less.render` compiles the first and refuses the second with `Unrecognised input`. Nothing is reported on the second run, since the rule got what it asked for. Less matches a flag with `` /^! *important/ ``, so a flag spelled in any other case is no flag to its expression reader, while `postcss-less` files every spelling in `raws.important` alike.

Measured against `less@4.9.1` over 25 value spellings × two property kinds, each run through `--fix` under `never` and both sides handed to `less.render`. Thirteen rows of an ordinary declaration compile before the fix and are refused after it:

| Value | Why the expression reader refuses it |
| --- | --- |
| `pink !IMPORTANT` | the flag is not the lower-case one Less reads |
| `!`, `?`, `%`, `&`, `<`, `=` | a lone character that is no entity |
| `a)b` | a closing parenthesis that opened nothing |
| U+000B, U+0085, U+00A0, U+2007, U+FEFF | a code point Less reads as neither whitespace nor a name |

`color: :` is the one such row the plugin already declines. The eleven rows that compile both ways — `pink`, `pink !important`, `a b c`, `rgb(0,0,0)`, `#fff`, `calc(1px + 2px)`, `"a"`, `a,b`, `[a]`, `-`, `local(a) format('b')` among them — are what the option is for, and they must keep their fix.

A custom property is not affected: its value goes to `permissiveValue(/[;}]/)`, which stops at the closing brace as readily as at the semicolon, so all 25 spellings compile with the semicolon and without it. That is why #358 lets a custom property keep its fix everywhere but the empty value.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/less/declaration-block-trailing-semicolon": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### What did you expect to happen?

The semicolon is taken away wherever Less parts with it, and the file still compiles.

### What actually happened?

The semicolon is taken away behind every value the plugin reads as a value, and the file stops compiling wherever Less's expression reader cannot read that value on its own.

### Do you have a proposal to fix the bug?

Whether `value()` reads a text is Less's expression grammar, and the plugin does not depend on Less, so nothing here can be settled by asking the compiler at lint time. Whether `anonymousValue()` matched settles nothing either: `pink` is admitted by it and read by `value()` as well, `pink !IMPORTANT` is admitted by it and refused by `value()`, so the same admission covers a row that keeps its fix and a row that must lose it.

That leaves declining for every declaration under Less, as the rule already declines for every at-rule Less reads as one — the safe direction, at the price of the option under Less — unless a narrower reading turns up that is a rule of Less's own rather than a guess. The `never` option is already a warning nobody can answer behind a Less at-rule, so the price is one this rule has paid before.

This is not #358's question, which is the value that is missing altogether. Found while writing the branch that closes issue #358, where the matrix above was measured.
